### PR TITLE
[5.6] Fix tests in FoundationTestResponseTest that can't fail

### PR DESCRIPTION
**Pull request by SjorsO**
#25232 added a few tests that can't fail because they do something like the following:
```php
try {
    $response->assertSeeTextInOrder(['baz', 'foobar']);

    $this->fail('Assertion was expected to fail.');
} catch (\PHPUnit\Framework\AssertionFailedError $e) {
    //
}
```

These tests will always pass because `$this->fail()` throws an exception which is caught by the surrounding catch.

The reason this happened is because the new tests in #25232 were copy and pasted from existing tests. But the existing tests only worked because they had their own tests: [SeeInOrderTest](https://github.com/laravel/framework/blob/0ffdf3ed8d7e26a4ce3f30167e3840e3780a62d5/tests/Foundation/Testing/Constraints/SeeInOrderTest.php)

This PR fixes the tests that can't fail, and cleans up the existing tests so the `SeeInOrderTest` becomes unnecessary.

### CODE REVIEW

**Comment by Juddling** [comment]
Much better!

Given that the PR is mainly to clean up where we expect exceptions to be thrown, I feel it reads nicer to use:

```php
$this->expectException(\PHPUnit\Framework\AssertionFailedError::class);
```

Then there is no need for the try/catch, PHPUnit will handle it all. If there is no exception, PHPUnit will given an error saying that it expected to see an exception thrown.

File: tests/Filesystem/tmp/foo.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+foo
```
Comment (jmarcher) [inline tests/Filesystem/tmp/foo.txt]: What is this file?

Comment (SjorsO) [reply tests/Filesystem/tmp/foo.txt]: Good catch, thank you

**Comment by SjorsO** [comment]
Good idea @Juddling, i've changed the tests to use `$this->assertException()`

**Pull request merged**